src/west/manifest.py

```diff
@@ -90,8 +90,8 @@
 class Submodule(NamedTuple):
     '''Represents a Git submodule within a project.'''
 
-    name: str
     path: str
+    name: str = ''
```

Comment:
```suggestion
    name: Optional[str] = None
```

Why not this instead? We can still write `if foo.name:`, but having a nullable type seems nicer to me.